src/libs/Navigation/linkingConfig/getAdaptedStateFromPath.ts

```diff
@@ -165,18 +171,25 @@ function getAdaptedState(state: PartialState<NavigationState<RootStackParamList>
 
         if (topmostNestedRHPRoute) {
             let matchingRootRoute = getMatchingRootRouteForRHPRoute(topmostNestedRHPRoute);
-
+            const isRHPScreenOpenedFromLHN = topmostNestedRHPRoute?.name && RHP_SCREENS_OPENED_FROM_LHN.includes(topmostNestedRHPRoute?.name as RHPScreenOpenedFromLHN);
```

Comment:
`Track expense` adds self DM as matching route and it should be added in the RHPScreensOpenedFromLHN. Fixed with #46848 . Issue here #46674.